samples/dfu/src/main.c

```diff
@@ -152,6 +152,7 @@ static void golioth_on_connect(struct golioth_client *client)
 
 	k_sem_give(&sem_connected);
 
+	memset(&update_ctx, 0, sizeof(update_ctx));
```

Comment:
`update_ctx` is accessed from `main()`, which is executed from another thread ("main" thread vs "golioth system client" thread). This means that accessing (in this case writing / clearing) it creates a race condition between threads and undefined behavior.